### PR TITLE
feat(payment module): setting orders and building the doctors seeder

### PR DESCRIPTION
**Pull request by JeanVittory**
# SUMMARY

The connection between backend and orders models when a product is bought is complete. A seeder to feed the doctor model was created

# TO-DO

keep building the payment module

### CODE REVIEW

File: src/api/controllers/Payments.controller.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import { Payments } from '../service/Payments.service';
+import { ApiError } from '../../config/middlewares/errorHandler/ApiError.middlewares';
+import PrismaError from '../../config/middlewares/errorHandler/PrismaErrorHandler.middleware';
+
+export class PaymentsController {
+  constructor() {}
+
+  static async checkout(req: Request, res: Response, next: NextFunction) {
+    try {
+      console.log('h3lo');
```
Comment (ht1204) [inline src/api/controllers/Payments.controller.ts]: what is this console.log doing?

File: src/api/middlewares/payments.middleware.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Request, Response, NextFunction } from 'express';
+import { ApiError } from '../../config/middlewares/errorHandler/ApiError.middlewares';
+import payments from '../schemas/payments.schema';
+
+const paymentsValidator = (req: Request, res: Response, next: NextFunction) => {
+  const { error } = payments.validate(req.body);
+  if (error) return next(ApiError.BadRequest('Error validating your bought'));
```
Comment (ht1204) [inline src/api/middlewares/payments.middleware.ts]: `Error validating your purchase`

File: src/api/service/Orders.service.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import { PrismaClient } from '@prisma/client';
+import { Order } from '../interfaces/Order.interface';
+
+const prisma = new PrismaClient();
+
+type id = {
```
Comment (ht1204) [inline src/api/service/Orders.service.ts]: `type IdentificationType` would be better named.

File: src/api/service/Orders.service.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+export class Orders {
+  constructor() {}
+
+  static async createOrder({ userId, cart, amount, card, state }: Order): Promise<any> {
+    try {
+      // TO-DO: map the appointments when the user try to schedule one
+      //const appointmentsId = cart.appointments.map((appointment) => ({ id: appointment.id }));
+      if (Object.keys(cart).includes('products') && !Object.keys(cart).includes('appointments')) {
+        const productsId = cart.products.map((product) => ({ id: product.id }));
+        const { id } = await prisma.order.create({
```
Comment (ht1204) [inline src/api/service/Orders.service.ts:20]: `const { products, brand, country, exp_month, exp_year, last4 } = cart;`

File: src/api/service/Payments.service.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+  static async processPayment({ paymentMethod, cart, amount, userId }: IPayment) {
+    try {
+      const { id } = paymentMethod;
+      const card = paymentMethod.card;
+      const response = await stripe.paymentIntents.create({
+        currency: 'USD',
+        payment_method: id,
+        amount: amount,
+      });
```
Comment (ht1204) [inline src/api/service/Payments.service.ts:21]: `const { status: state, id: idPayment } = await stripe.paymentIntents.create({...`

File: src/api/service/Payments.service.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+      const order = {
+        userId,
+        cart,
+        amount,
+        card,
+        state: response.status,
+      };
+      const { id: orderId } = await Orders.createOrder(order);
+      if (orderId) {
+        const { status } = await stripe.paymentIntents.confirm(response.id);
```
Comment (ht1204) [inline src/api/service/Payments.service.ts]: `const { status } = await stripe.paymentIntents.confirm(idPayment);`

File: src/api/controllers/Payments.controller.ts (hunk 1 above)
Comment (JeanVittory) [reply src/api/controllers/Payments.controller.ts]: done

File: src/api/middlewares/payments.middleware.ts (hunk 2 above)
Comment (JeanVittory) [reply src/api/middlewares/payments.middleware.ts]: done

File: src/api/service/Orders.service.ts (hunk 3 above)
Comment (JeanVittory) [reply src/api/service/Orders.service.ts]: done

File: src/api/service/Orders.service.ts (hunk 4 above)
Comment (JeanVittory) [reply src/api/service/Orders.service.ts:20]: done

File: src/api/service/Payments.service.ts (hunk 5 above)
Comment (JeanVittory) [reply src/api/service/Payments.service.ts:21]: done

File: src/api/service/Payments.service.ts (hunk 6 above)
Comment (JeanVittory) [reply src/api/service/Payments.service.ts]: done

**Review by ht1204 — APPROVED** [review]

**Pull request merged**